### PR TITLE
Update text domain to eight-day-week-print-workflow

### LINKED ISSUE #40: Not properly prepared for translations
**Describe the bug**
https://translate.wordpress.org/projects/wp-plugins/eight-day-week-print-workflow/ states:
> This plugin is not properly prepared for localization.
Further research from [Slack](https://wordpress.slack.com/archives/C0E7F4RND/p1564518145101300) shows:
> Plugin is not compatible with language packs: Wrong text domain in header.

**Expected behavior**
- [ ] Translations are open for contributions.
- [ ] language packs are properly available when translations are complete within a specific language

### PR DESCRIPTION
**Pull request by jeffpaul**
### Description of the Change

The Text Domain is currently set to `eight-day-week`, but needs to be `eight-day-week-print-workflow` to match the plugin slug on WordPress.org in order for translations to be open on translate.wordpress.org.  This PR updates the Text Domain to `eight-day-week-print-workflow`.

### Alternate Designs

none

### Benefits

Translations will now be open for contributions on translate.wordpress.org.

### Possible Drawbacks

none identified

### Verification Process

Manually verified via GitHub web UI.

### Checklist:

- [ ] I have read the [**CONTRIBUTING**](https://github.com/10up/eight-day-week/blob/develop/CONTRIBUTING.md) document.
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my change.
- [ ] All new and existing tests passed.

### Applicable Issues

Relates to #40.

### CODE REVIEW

**Comment by jeffpaul** [comment]
I suspect `eight-day-week-de_DE.mo` might also need an edit for the .org slug but I'm unsure whether that filename also needs to be updated (same for the PO and POT files).  I'm going to read some more on .org translations before this is likely ready for review.

**Comment by jeffpaul** [comment]
So I _think_ that captures the various changes called out as needed on https://developer.wordpress.org/plugins/internationalization/how-to-internationalize-your-plugin/.  Certainly needs reviewing to see if anything was missed or incorrectly updated.  Fun.

File: includes/functions/core.php [hunk 1]
```diff
@@ -44,9 +44,9 @@ function init() {
 }
 
 function i18n() {
-	$locale = apply_filters( 'plugin_locale', get_locale(), 'eight-day-week' );
-	load_textdomain( 'eight-day-week', WP_LANG_DIR . '/print-production/print-production-' . $locale . '.mo' );
-	load_plugin_textdomain( 'eight-day-week', false, plugin_basename( EDW_PATH ) . '/languages/' );
+	$locale = apply_filters( 'plugin_locale', get_locale(), 'eight-day-week-print-workflow' );
+	load_textdomain( 'eight-day-week-print-workflow', WP_LANG_DIR . '/eight-day-week-print-workflow/eight-day-week-print-workflow-' . $locale . '.mo' );
```
Comment (jeffpaul) [inline includes/functions/core.php]: I'm least confident in this change here, so confirmation that the concatenation of this all makes sense would be good.

**Comment by adamsilverstein** [comment]
@jeffpaul 	I added a few missing textdomain replacements and renamed the translation files. 

### Testing 
I tested this locally by setting my profile language to German and checking that all our custom strings are properly translated. They all looked good, other than a few that are hard coded into our JavaScript (for example https://github.com/10up/eight-day-week/blob/develop/assets/js/src/scripts.js#L356) - these can wither use JS translation functions or be localized from PHP and can be handled in a follow up PR.

File: includes/functions/core.php (hunk 1 above)
Comment (adamsilverstein) [reply includes/functions/core.php]: I think this is right, it works correctly locally. I can check some other plugins as well to confirm the approach.

**Review by adamsilverstein — DISMISSED** [review]
Swell

File: includes/functions/core.php (hunk 1 above)
Comment (helen) [reply includes/functions/core.php]: I don't think we need this anymore if trying to use WordPress.org language packs? It means bumping the WP required version (4.6+ I believe) and then we can remove the text domain header. I'm going to ask somebody who knows more about how the .org side about why this doesn't work with the custom text domain header set.

Comment (jeffpaul) [reply includes/functions/core.php]: Sounds good, not super urgent to get these changes handled, just more a "nice to check it off the list" sort of thing.

**Comment by jeffpaul** [comment]
Updated "Requires at least" to version 4.6, this should be the [last task](https://github.com/10up/eight-day-week/pull/41/#issuecomment-525370277) to have this PR complete to allow proper translations.  Based on @adamsilverstein's [prior approval](https://github.com/10up/eight-day-week/pull/41/#pullrequestreview-272724940) and final tweaks, I'll merge and include this alongside the "Tested up to" bump to version 5.3.

**Pull request merged**